firebase-firestore/src/main/java/com/google/firebase/firestore/core/Target.java

```diff
@@ -153,23 +152,21 @@ public boolean hasLimit() {
    * Returns the list of values that are used in != or NOT_IN filters. Returns {@code null} if there
    * are no such filters.
    */
-  public @Nullable List<Value> getNotInValues(FieldIndex fieldIndex) {
-    List<Value> values = new ArrayList<>();
-
+  public @Nullable Collection<Value> getNotInValues(FieldIndex fieldIndex) {
```

Comment:
This method now supports a == 1 && a != 2, which now correctly just returns a != 2 instead of the concatenation of both.